File: MyBusMap/src/main/java/mk/com/finki/mybusmap/service/BusScheduleService.java
package mk.com.finki.mybusmap.service;

import mk.com.finki.mybusmap.model.BusSchedule;
import mk.com.finki.mybusmap.model.dto.BusScheduleDto;

import java.util.List;

public interface BusScheduleService {
    List<BusSchedule> getAllBusSchedules();
    BusSchedule createBusSchedule(BusScheduleDto busScheduleDto);
    BusSchedule getBusScheduleById(Long id);
    BusSchedule updateBusSchedule(Long id, BusScheduleDto busScheduleDto);
    void deleteBusSchedule(Long id);
    List<BusSchedule> getBusSchedulesByBusStopId(Long busStopId);
    List<BusSchedule> findAllByArrivalTimeAndBusStop_Id(String arrivalTime, Long busStopId);
    List<BusSchedule> getBusSchedulesByBusId(Long busId);

}

File: frontend/src/api/busScheduleService.js
import api from './axiosConfig';

export const getAllBusSchedules = async () => {
    const response = await api.get('/api/bus-schedule');
    return response.data;
};

export const getBusScheduleById = async (id) => {
    const response = await api.get(`/api/bus-schedule/${id}`);
    return response.data;
};

export const createBusSchedule = async (busScheduleDto) => {
    const response = await api.post('/api/bus-schedule/add', busScheduleDto);
    return response.data;
};

export const updateBusSchedule = async (id, busScheduleDto) => {
    const response = await api.post(`/api/bus-schedule/edit/${id}`, busScheduleDto);
    return response.data;
};

export const deleteBusSchedule = async (id) => {
    await api.delete(`/bus-schedule/delete/${id}`);
};

export const getBusSchedulesByBusStopId = async (busStopId) => {
    const response = await api.get(`/api/bus-schedule/bus-stop/${busStopId}`);
    return response.data;
};

export const getBusSchedulesByBusId = async (busId) => {
    const response = await api.get(`/api/bus-schedule/bus/${busId}`);
    return response.data;
};

export const findSchedulesByArrivalTime = async (arrivalTime, busStopId) => {
    const response = await api.get(`/api/bus-schedule/arrival-time/bus-stop/${busStopId}`, {
        params: { arrivalTime }
    });
    return response.data;
};
